libraries/chain/include/eosio/chain/database_utils.hpp

```diff
@@ -195,7 +183,74 @@ namespace fc {
    void from_variant( const variant& v, eosio::chain::shared_vector<T>& sv ) {
       std::vector<T> _v;
       from_variant(v, _v);
-      sv = eosio::chain::shared_vector<T>(_v.begin(), _v.end(), sv.get_allocator());
+      sv = v;
+   }
+}
+
+namespace fc::raw {
+   // pack/unpack chainbase::shared_cow_string
+   // ----------------------------------------
+   template<typename Stream>
+   inline void pack(Stream& s, const chainbase::shared_string& v) {		
+      FC_ASSERT(v.size() <= MAX_SIZE_OF_BYTE_ARRAYS);		
+      fc::raw::pack(s, fc::unsigned_int((uint32_t)v.size()));		
+      if (v.size())
+         s.write(v.data(), v.size());		
+   }		
+
+   template<typename Stream>
+   inline void unpack(Stream& s, chainbase::shared_string& v) {
+      fc::unsigned_int sz;
+      fc::raw::unpack(s, sz);
+      FC_ASSERT(sz.value <= MAX_SIZE_OF_BYTE_ARRAYS);
```

Comment:
This is more strict than what existed previously, but so far I can't see any reason it'd be a problem..